layer/layer.go

```diff
@@ -168,6 +177,7 @@ type MountInit func(root string) error
 // read-only and read-write layers.
 type Store interface {
 	Register(io.Reader, ChainID) (Layer, error)
```

Comment:
Rather than create a separate method, `Register` should just take a descriptor. If no packed digest is available, it should be omitted.